pkg/plugins/golang/v4/scaffolds/internal/templates/test/e2e/suite.go

```diff
@@ -100,20 +100,20 @@ var _ = BeforeSuite(func() {
 	// we check for its presence before execution.
 	// Setup CertManager before the suite if not skipped and if not already installed
 	if !skipCertManagerInstall {
-		By("checking if cert manager is installed already")
-		isCertManagerAlreadyInstalled = utils.IsCertManagerCRDsInstalled()
-		if !isCertManagerAlreadyInstalled {
+		By("checking if cert manager is installed already") 
+		if !utils.IsCertManagerCRDsInstalled() {
 			_, _ = fmt.Fprintf(GinkgoWriter, "Installing CertManager...\n")
 			Expect(utils.InstallCertManager()).To(Succeed(), "Failed to install CertManager")
+			certManagerWasInstalledByTest = true
```

Comment:
What problem does this change solve? Aren’t we doing exactly the same thing as before?

In the original code, the flag defaults to false (so we don’t skip the install).
If we detect cert-manager CRDs on the cluster (regardless of how cert-manager was installed), we set it to true and skip the install.

So functionally, the behavior seems unchanged.

Reply:
@camilamacedo86 if you have a better/safer way to fix the problem then we should do that. I am not insisting on this specific solution.

however as the code stands today, there is a serious bug.